app/workloadSummaryReport/services/workloadSummaryActions.js

```diff
@@ -0,0 +1,562 @@
+
+										sectionGroup.maxEnrollment = sectionGroup.maxEnrollment || 0;
+										sectionGroup.actualEnrollment = sectionGroup.actualEnrollment || 0;
+										sectionGroup.previousEnrollment = sectionGroup.previousEnrollment || 0;
+
+										if (sectionGroupUniqueKey == censusSectionGroupKey) {
+											if (isPreviousYear) {
+												sectionGroup.previousEnrollment += censusSection.currentEnrolledCount;
+											} else {
+												sectionGroup.actualEnrollment += censusSection.currentEnrolledCount;
+
+												_self._getSectionsForSectionGroup(sectionGroup).forEach(function(section) {
+													sectionGroup.maxEnrollment += section.seats;
+												});
+											}
+										}
+									});
+								}
+							});
+
+							completedCalls += 1;
+
+							if (openCalls == completedCalls) {
+								_self._performCalculations();
+							}
+						}, function (err) {
+							$rootScope.$emit('toast', { message: "Could not retrieve enrollment data.", type: "ERROR" });
+						});
+					});
+				});
+			},
+			_getScheduleSubjectCodes: function() {
+				var subjectCodes = [];
+				var sectionGroups = WorkloadSummaryReducers._state.sectionGroups;
+				var courses = WorkloadSummaryReducers._state.courses;
+
+				sectionGroups.ids.forEach(function(sectionGroupId) {
+					var sectionGroup = sectionGroups.list[sectionGroupId];
+					var subjectCode = courses.list[sectionGroup.courseId].subjectCode;
+
+					if (subjectCodes.indexOf(subjectCode) == -1) {
+						subjectCodes.push(subjectCode);
+					}
+				});
+
+				return subjectCodes;
+			},
+			_getScheduleTermCodes: function(isPreviousYear) {
+				var termCodes = [];
+
+				var sectionGroups = WorkloadSummaryReducers._state.sectionGroups;
+
+				sectionGroups.ids.forEach(function(sectionGroupId) {
+					var termCode = sectionGroups.list[sectionGroupId].termCode;
+
+					// Get the previous year's termCode instead
+					if (isPreviousYear) {
+						var year = parseInt(termCode.substring(0,4)) - 1;
+						var term = termCode.substring(4,6);
+						var previousYearTermCode = year + term;
+						if (termCodes.indexOf(previousYearTermCode) == -1) {
+							termCodes.push(previousYearTermCode);
+						}
+					} else {
+						if (termCodes.indexOf(termCode) == -1) {
+							termCodes.push(termCode);
+						}
+					}
+				});
+
+				return termCodes;
+			},
+			_getSectionsForSectionGroup: function (sectionGroup) {
+				var sectionGroups = WorkloadSummaryReducers._state.sectionGroups;
+				var sections = WorkloadSummaryReducers._state.sections;
+
+				var matchingSections = [];
+				sections.ids.forEach(function(sectionId) {
+					var section = sections.list[sectionId];
+					if (section.sectionGroupId == sectionGroup.id) {
+						matchingSections.push(section);
+					}
+				});
+
+				return matchingSections;
+			},
+			_orderInstructorTypeIdsAlphabetically: function (instructorTypeIds) {
+				var alphabeticalOrder = [3, 5, 1, 7, 6, 8, 4, 2];
```

Comment:
A magic array appears! If this is absolutely necessary (why not sort the labels once they're text?), please leave a comment indicating where this order comes from and why it isn't subject to change in the future when new instructor types are added/removed.

Reply:
Abstracting to support new types would be good, and this would be a good candidate for termService as I think we use this 'alphabetical' ordering concept elsewhere, will refactor.